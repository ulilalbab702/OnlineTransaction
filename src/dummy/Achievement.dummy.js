export const Achievement = [
    {
        "Code": "FM-JBYPM",
        "Name": "FMC Jembayan PAMA",
        "Percent": 50,
        "BackLogOpen": 1,
        "BackLogClose": 1,
        "Priority1": 1,
        "Priority2": 0,
        "Priority3": 1,
        "Priority4": 0
    }
]