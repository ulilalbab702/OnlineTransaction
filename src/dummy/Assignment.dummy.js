export const listOfMechanicDummy = [
    {
      "Username": "string",
      "FirstName": "string",
      "LastName": "string",
      "Email": "string",
      "MechanicUT": true,
      "Jobs": [
        {
          "JobType": "string",
          "Total": 0
        }
      ]
    }
  ]
export const jobSummaryDummy = {
  "Date": "2020-06-04T16:18:05.5472055",
  "CreatedBy": "AGUS LISANTO",
  "AssignedBy": "AGUS LISANTO",
  "MechanicLeader": {
      "UserName": "82107038@unitedtractors.com",
      "Name": "SUPRIYADI SUPRIYADI"
  },
  "Mechanics": [],
  "HandoverDate": null,
  "HandoverCreatedBy": null,
  "HandoverAssignedBy": null,
  "HandoverMechanicLeader": null,
  "HandoverMechanics": null,
  "Job": {
      "WoId": "8ff577d1-2911-4bbd-9d55-ffad4fe76fc5",
      "WoNumber": "71018650",
      "JobType": "INS",
      "UnitModel": "HD465-7",
      "SerialNumber": "15163",
      "UnitCode": "HT3022",
      "EquipmentNumber": "71506",
      "IsHandover": false,
      "Id": "a162bd06-6aca-499a-9f97-a5839b8a0612",
      "CreatedBy": "80107002@unitedtractors.com",
      "CreatedDate": "2020-06-04T16:18:05.5472055",
      "UpdatedBy": "80107002@unitedtractors.com",
      "UpdatedDate": "2020-06-04T16:18:05.5472055",
      "IsActive": true
  }
}