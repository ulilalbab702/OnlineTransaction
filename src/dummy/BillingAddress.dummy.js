export const dataBillingAddressDummy = [
    {
        "fullName": "Test Integrasi",
        "contactNumber": "088122345",
        "email": "testintegrasi@unitedtractors.com",
        "position": "string",
        "addressLabel": "Kantor PT Global Service Indonesia",
        "city": "Jakarta",
        "address": "Jl Raya kaliabang Tengah, Ruko San Jaya, Blok A5 No 9,\r\nSebalah Indomaret",
        "postalCode": "13920",
        "locationLatitude": "3.669276961596295",
        "locationLongitude": "102.4003031491451",
        "isPrimary": true
    }]