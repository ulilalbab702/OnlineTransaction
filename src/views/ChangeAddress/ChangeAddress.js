import React, { useEffect, useState } from "react";
import "./ChangeAddress.scss";
import { Row, Col } from "reactstrap";
import firebase from "../../firebase/firebase";
import ModalAddress from "../../components/ModalAddAddress/ModalAddAddress";
import ModalEditAddress from "../../components/ModalEditAddress/ModalEditAddress";
import Breadcrumb from '../../components/Breadcrumb/Breadcrumb';
import ModalSuccessAddress from "components/ModalSuccessAddress/ModalSuccessAddress";
import { MENU } from "constants/menu";


const ChangeAddress = (props) => {
  const [modalAddress, setModal] = useState(false);
  const [modalEditAddress, setModalEditAddress] = useState(false);
  const [successAddAddress, setSuccessAddAddress] = useState(false);
  const [successEditAddress, setSuccessEditAddress] = useState(false);
  const [address, selectAddress] = useState(null);
  const [districtList, setDistrictList] = useState(null);
  const [postalcodeList, setPostalcodeList] = useState(null);
  const [keyword, setKeyword] = useState("");
  const [idKeyword, setIdKeyword] = useState("");
  const [postalCode, setPostalCode] = useState("");
  const [dataEditAddress, setDataEditAddress] = useState(null);
  const [dataCity, setDataCity] = useState(null);
  const [dataProvince, setDataProvince] = useState(null);
  const [dataVillages, setDataVillages] = useState(null);
  const [dataPostalCodes, setDataPostalCodes] = useState(null);

  const setAnalyticClevertap = (action, event, screen, product) => {
    if (props.user.user) {
      const { userName } = props.user.user;
      if (product !== null) {
        window.clevertapEventProduct(`${product[0] + "_LogedIn"}`, product[1]);
      } else {
        window.clevertapEvent(`${event + "_LogedIn"}`, props.user.user);
      }
      firebase.analytics().setUserId(userName);
      firebase.analytics().setUserProperties(userName, "username:" + userName + `||View: ${screen}`);
      firebase.analytics().setCurrentScreen(screen, screen);
      firebase.analytics().logEvent(`${event + "_LogedIn"}`);
      firebase.analytics().setUserProperties("username", userName);
      firebase.analytics().setUserProperties(action, `${event + "_LogedIn"}`);
    }
  };

  useEffect(() => {
    setAnalyticClevertap("view", "View_ChangeAddress_Screen", "View_ChangeAddress_Screen", null);
    window.scrollTo(0, 0);
  }, []);

  useEffect(() => {
    const { user } = props
    const accessToken = user.user.tokenResponse.accessToken
    const userId = user.user.userId
    async function fetchData() {
      const response = await props.fetchGetBillingList(userId, accessToken);
    }
    fetchData();
  }, [address, props.dataPut, props.dataPostBilling]);

  useEffect(() => {
    if (keyword.length >= 3) {
      async function fetchSearch() {
        await props.fetchSearchDistrict(keyword);
      }
      fetchSearch();
    } else {
      setDistrictList(null)
      props.fetchResetDistrict();
    }
  }, [keyword, idKeyword]);

  useEffect(() => {
    if (postalCode.length == 5) {
      async function fetchPostalCode() {
        await props.fetchSearchPostalCode(postalCode);
      }
      fetchPostalCode();
    } else {
      setPostalcodeList(null)
      props.fetchResetPostalcode();
    }
  }, [postalCode]);

  useEffect(() => {
    if (props.districtList) {
      setDistrictList(props.districtList)
    } else {
      setDistrictList(props.districtList)
    }
    if (props.postalcodeList) {
      setPostalcodeList(props.postalcodeList)
    } else {
      setPostalcodeList(props.postalcodeList)
    }
    setDataCity(props.dataCity)
    setDataProvince(props.dataProvince)
    setDataVillages(props.dataVillages)
    setDataPostalCodes(props.dataPostalCodes)
  }, [props.districtList, props.postalcodeList, props.dataCity, props.dataProvince, props.dataPostalCodes]);

  useEffect(() => {
    var err = String(props.billingAddressListError)
    if (err.includes('401')) {
    }
  })

  useEffect(() => {
    if (props.dataPostBilling?.status) {
      if (props.dataPostBilling?.status == 200) {
        setSuccessAddAddress(true);
        setAnalyticClevertap("action", "Action_AddBillingAddress_Success", "View_ChangeAddress_Screen", null);
      } else if (props.errPostBilling != null) {
        setAnalyticClevertap("action", "Action_AddBillingAddress_Error", "View_ChangeAddress_Screen", null);
      }
    }
  }, [props.dataPostBilling, props.errPostBilling])

  useEffect(() => {
    if (props.dataPut?.status == 200) {
        setSuccessEditAddress(true);
        setAnalyticClevertap("action", "Action_EditBillingAddress_Success", "View_UserProfile_Screen", null);
    } else if (props.errPut != null) {
        setAnalyticClevertap("action", "Action_EditBillingAddress_Error", "View_UserProfile_Screen", null);
    }
}, [props.dataPut])

  const _resetData = () => {
    setPostalCode("")
    setKeyword("")
    setDistrictList(null)
    setPostalcodeList(null)
    setModal(false)
    setModalEditAddress(false)
    props.fetchResetPostalcode();
    props.fetchResetDistrict();
    setDataEditAddress(null)
  }

  const _resetAll = () => {
    setPostalCode("")
    setKeyword("")
    setDistrictList(null)
    setPostalcodeList(null)
    props.fetchResetPostalcode();
    props.fetchResetDistrict();
    setDataEditAddress(null)
  }


  const _handleChangeProvince = (value) => {

  }

  const _handleChangeCity = (value) => {
    setKeyword(value);
    setIdKeyword(null);
    props.fetchResetDistrict();
    setPostalcodeList(null)
  };

  const _handleChangePostalCode = (event) => {
    setPostalCode(event);
    props.fetchResetPostalcode();
  }

  const _handleSubmit = async (data) => {
    if (data) {
      const { user } = props
      const accessToken = user.user.tokenResponse.accessToken
      const userId = user.user.userId
      await props.fetchPostBillingAddress(accessToken, userId, data);
    }
  }

  const _handleSubmitEdit = async (data) => {
    if (data) {
      const { user } = props
      const accessToken = user.user.tokenResponse.accessToken
      await props.fetchPutBillingList(accessToken, data.billingId, data, false);
    }
  }

  const handleDelete = async (data) => {
    if (data) {
      const { user } = props
      const accessToken = user.user.tokenResponse.accessToken
      await props.fetchDeleteBillingList(accessToken, data.billingId);        
      setAnalyticClevertap("click", "Click_DeleteBillingAddress", "View_UserProfile_Screen", null);
    }
  }

  const actionSelect = async (item) => {
    const { user } = props
    const accessToken = user.user.tokenResponse.accessToken
    selectAddress(item.billingId)
    await props.fetchPutBillingList(accessToken, item.billingId, item, true);        
    setAnalyticClevertap("click", "Click_SelectBillingAddress", "View_UserProfile_Screen", null);
  }

  const clickEdit = (item) => {
    setAnalyticClevertap("click", "Click_ModalEditAddress", "View_ChangeAddress_Screen", null);
    props.fetchResetDistrict();
    const dataEdit = {
      billingId: item.billingId,
      fullName: item.fullName,
      contactNumber: item.contactNumber,
      addressLabel: item.addressLabel,
      city: item.city,
      province: item.province,
      districts: item.districts,
      address: item.address,
      postalCode: item.postalCode,
      isPrimary: item.isPrimary,
      village: item.village,
    }
    async function fetchSearch() {
      await props.fetchSearchDistrict(item.city);
    }
    fetchSearch();
    setDataEditAddress(dataEdit)
    setModalEditAddress(true)
  }

  const clickConfirmAddress = () => {
    setAnalyticClevertap("click", "Click_ConfirmAddress", "View_ChangeAddress_Screen", null);
    const { state } = props.location
    props.push(MENU.BILLING, state)
  }

  const _renderBreadcrumb = () => {
    const breadcrums = [
      {
        'url': MENU.HOME,
        'name': 'Part Online Transaction'
      }
    ];
    return (
      <Breadcrumb
        linkBreadcrumb={breadcrums}
        typography={"Alamat"}
      />
    )
  };

  const _renderModalSuccessAddAddress = () => {
    return (
      <ModalSuccessAddress
        isOpen={successAddAddress}
        isClose={() => setSuccessAddAddress(false)}
        title="Hore Selamat..."
        description="Alamat berhasil ditambahkan"
      />
    );
  };

  const _renderModalSuccessEditAddress = () => {
    return (
      <ModalSuccessAddress
        isOpen={successEditAddress}
        isClose={() => setSuccessEditAddress(false)}
        title="Hore Selamat..."
        description="Alamat berhasil diubah"
      />
    );
  };

  return (
    <div>
      {_renderModalSuccessAddAddress()}
      {_renderModalSuccessEditAddress()}
      <Col className='bgCA'>
        {_renderBreadcrumb()}
        <p className='addressTitleCA'>Alamat Tersimpan</p>
        <hr className='lineAddress' />
        {props.billingAddressList &&
          props.billingAddressList.length > 0 &&
          props.billingAddressList.map((item) => {
            return (
              <div key={item.billingId}>
                {item.isPrimary ? (
                  <div className='addressBoxBillingSelected'>
                    <p className='address1'>{item.addressLabel}</p>
                    <p className='address2'>{item.fullName} ({item.contactNumber})</p>
                    <p className='address3'>{item.address}</p>
                    <Row className='m-0 rowActionAddress'>
                      <p className='address4' onClick={() => clickEdit(item)}>Ubah Alamat</p>
                    </Row>
                  </div>
                ) : (
                  <div className='addressBoxBilling'>
                    <p className='address1'>{item.addressLabel}</p>
                    <p className='address2'>{item.fullName} ({item.contactNumber})</p>
                    <p className='address3'>{item.address}</p>
                    <Row className='m-0 rowActionAddress'>
                      <p className='address4' onClick={() => clickEdit(item)} >Ubah Alamat</p>
                      <p onClick={() => actionSelect(item)} className='address5'>Pilih Alamat</p>
                      <p className='address5' onClick={() => handleDelete(item)} style={{ backgroundColor: '#D0021B' }}>Hapus Alamat</p>
                    </Row>
                  </div>
                )}
              </div>
            );
          })}
        {props.billingAddressList &&
          props.billingAddressList.length > 0 &&
          props.billingAddressList.filter(item => item.isPrimary).length > 0 ?
          (
            <p className='confirmAddress' onClick={() => clickConfirmAddress()}>Konfirmasi Alamat</p>
          ) : (
            <p className='confirmAddress' style={{ backgroundColor: "#d8d8d8", cursor: "default" }}>Konfirmasi Alamat</p>
          )
        }
        <hr className='lineAddress' />
        <p
          className='addAddress'
          onClick={() => {
            setAnalyticClevertap("click", "Click_AddBillingAddress", "View_ChangeAddress_Screen", null);
            setModal(!modalAddress);
          }}
        >
          + Tambah Alamat Baru
        </p>
        <ModalAddress
          isOpen={modalAddress}
          isClose={() => _resetData()}
          name={keyword}
          dataDistrict={districtList}
          dataPostalCode={postalcodeList}
          dataCity={dataCity}
          dataProvince={dataProvince}
          dataVillages={dataVillages}
          dataPostalCodes={dataPostalCodes}
          resetAll={() => _resetAll()}
          handleChangeProvince={(value) => _handleChangeProvince(value)}
          handleChangeCity={(event) => _handleChangeCity(event)}
          handleChangePostalCode={(event) => _handleChangePostalCode(event)}
          handleSubmit={(data) => _handleSubmit(data)}
        />
        <ModalEditAddress
          isOpen={modalEditAddress}
          isClose={() => _resetData()}
          dataEditAddress={dataEditAddress}
          dataDistrict={districtList}
          dataPostalCode={postalcodeList}
          dataCity={dataCity}
          dataProvince={dataProvince}
          dataVillages={dataVillages}
          dataPostalCodes={dataPostalCodes}
          resetAll={() => _resetAll()}
          handleChangeProvince={(value) => _handleChangeProvince(value)}
          handleChangeCity={(event) => _handleChangeCity(event)}
          handleChangePostalCode={(event) => _handleChangePostalCode(event)}
          handleSubmit={(data) => _handleSubmitEdit(data)}
        />
      </Col>
    </div>
  );
};
export default ChangeAddress;
