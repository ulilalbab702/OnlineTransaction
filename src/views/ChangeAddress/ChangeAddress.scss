.bgCA {
  padding-top: 7vw;
  padding-left: 10vw;
  padding-right: 35vw;
}
.addressTitleCA {
  font-family: "SFProText-Bold";
  font-style: normal;
  font-size: 1.5vw;
  line-height: 2vw;
  margin-top: 2vw;
  color: #000000;
}
.lineAddress {
  flex: 1;
}
.addressBoxBilling {
  padding: 1vw;
  padding-left: 2.5vw;
  flex: 1;
  max-height: 14.6vw;
  background: #ffffff;
  border: 0.07vw solid #d4d5d5;
  box-sizing: border-box;
  border-radius: 1.25vw;
  margin-bottom: 1.5vw;
  overflow: auto;
}
.addressBoxBillingSelected {
  padding: 1vw;
  padding-left: 2.5vw;
  flex: 1;
  max-height: 14.6vw;
  background: #f9f9f9;
  border: 0.07vw solid #148f0f;
  box-sizing: border-box;
  border-radius: 1.25vw;
  margin-bottom: 1.5vw;
  overflow: auto;
}
.address1 {
  font-family: "SFProText-Bold";
  font-style: normal;
  font-weight: bold;
  font-size: 1.4vw;
  line-height: 21px;
  color: #050505;
}
.address2 {
  font-family: "SFProText-Regular";
  font-style: normal;
  font-size: 1.2vw;
  line-height: 2vw;
  letter-spacing: 0.01em;
  color: #000000;
  opacity: 0.4;
  margin-bottom: 0px;
}
.address3 {
  font-family: "SFProText-Regular";
  font-style: normal;
  font-size: 1.2vw;
  line-height: 2vw;
  letter-spacing: 0.01em;
  color: #000000;
  opacity: 0.4;
}
.rowActionAddress {
  display: flex;
  flex-direction: row;
  align-items: center;
}
.address4 {
  font-family: "SFProText-Bold";
  font-style: normal;
  font-size: 1.25vw;
  line-height: 1.4vw;
  color: #148f0f;
  opacity: 0.75;
  margin: 0;
  cursor: pointer;
}
.address5 {
  font-family: "SFProText-Bold";
  font-style: normal;
  font-weight: 600;
  font-size: 16px;
  line-height: 19px;
  text-align: center;
  letter-spacing: 0.01em;
  color: #ffffff;
  background: #5acc4d;
  border-radius: 8px;
  padding: 10px;
  margin: 0;
  margin-left: 20px;
  cursor: pointer;
}
.confirmAddress {
  width: 25.8vw;
  height: 3.7vw;
  font-family: "SFProText-Bold";
  font-style: normal;
  font-size: 1vw;
  line-height: 1.3vw;
  color: #000000;
  background-color: #ffd500;
  border-radius: 0.6vw;
  padding-top: 0.7vw;
  padding-bottom: 0.7vw;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 2.3vw;
  margin-left: auto;
  cursor: pointer;
}
.addAddress {
  height: 5vw;
  border: 0.2vw dashed #565555;
  text-align: center;
  box-sizing: border-box;
  border-radius: 1.25vw;
  font-family: "SFProText-Bold";
  font-style: normal;
  font-size: 1.4vw;
  line-height: 1.6vw;
  color: #050505;
  padding-top: 0.7vw;
  padding-bottom: 0.7vw;
  margin-bottom: 5vw;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
}
.modal-address {
  display: flex;
  justify-content: center;
  align-items: center;
  width: fit-content;
  height: fit-content;
  margin: auto;
  padding: 0;
  background-color: #ffffff;
}

@media only screen and (min-width: 1280px) {
  .paddingTP {
    padding-top: 6vw;
    padding-left: 10vw;
    padding-right: 30vw;
    margin-bottom: 10rem;
  }
}
