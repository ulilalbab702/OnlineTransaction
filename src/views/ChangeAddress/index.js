import ChangeAddress from './ChangeAddress';
import { connect } from 'react-redux';
import { BILLING_ADDRESS_REQ } from 'actions/actionTypes';
import { login } from 'actions/user';
import { push } from 'connected-react-router';
import {
    createLoadingSelector,
} from 'utils/selector.helper'
import { listBillingAddressGet, listBillingAddressPut, billingAddressPost, getSearchDistrict, getSearchPostalCode, resetDistrict, resetPostalcode, billingAddressDelete } from 'actions/billingAddress';

//Set selector loading 
const loadingSelector = createLoadingSelector([BILLING_ADDRESS_REQ]);
const mapStateToProps = state => ({
    isLoading: loadingSelector(state),
    billingAddressList: state.billingAddress.billingAddress.data,
    billingAddressListError: state.billingAddress.billingAddress.err,
    user: state.user,
    districtList: state.billingAddress.district.districtList,
    dataProvince: state.billingAddress.district.dataProvince,
    dataCity: state.billingAddress.district.dataCity,
    dataVillages: state.billingAddress.district.dataVillages,
    dataPostalCodes: state.billingAddress.district.dataPostalCodes,
    postalcodeList: state.billingAddress.district.postalcodeList,
    dataPut: state.billingAddress.billingAddress.dataPut,
    errPut: state.billingAddress.billingAddress.errPut,
    errPostBilling: state.billingAddress.billingAddress.errPostBilling,
    dataPostBilling: state.billingAddress.billingAddress.dataPostBilling
});

const mapDispatchToProps = dispatch => ({
    login: (data) => dispatch(login(data)),
    push: (url, state) => dispatch(push(url, state)),
    fetchGetBillingList: (userId, token) => dispatch(listBillingAddressGet(userId, token)),
    fetchPutBillingList: (token, billingId, data, changeStatus) => dispatch(listBillingAddressPut(token, billingId, data, changeStatus)),
    fetchDeleteBillingList: (token, billingId) => dispatch(billingAddressDelete(token, billingId)),
    fetchPostBillingAddress: (token, userId, data) => dispatch(billingAddressPost(token, userId, data)),
    fetchSearchDistrict: (city) => dispatch(getSearchDistrict(city)),
    fetchSearchPostalCode: (postalcode) => dispatch(getSearchPostalCode(postalcode)),
    fetchResetDistrict: () => dispatch(resetDistrict()),
    fetchResetPostalcode: () => dispatch(resetPostalcode()),
});

export default connect(mapStateToProps, mapDispatchToProps)(ChangeAddress);