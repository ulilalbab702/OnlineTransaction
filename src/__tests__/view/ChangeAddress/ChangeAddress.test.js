import React from "react";
import toJson from "enzyme-to-json";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import ChangeAddressPage from "views/ChangeAddress/ChangeAddress";
import { dataBillingAddressDummy } from "../../../dummy/ChangeAddress.dummy";

Enzyme.configure({ adapter: new Adapter() });
describe("Change Address Component", () => {
	const wrapper = shallow(
		<ChangeAddressPage 
			billingAddressList={dataBillingAddressDummy}
		/>
	);
	describe("Change Address Component Component", () => {
		it("Should match to snapshot", () => {
			expect(toJson(wrapper)).toMatchSnapshot();
		});
	});
});
