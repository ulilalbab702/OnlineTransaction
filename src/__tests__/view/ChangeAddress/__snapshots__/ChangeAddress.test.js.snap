// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Change Address Component Change Address Component Component Should match to snapshot 1`] = `
<div>
  <Col
    className="bgTP"
    lg="6"
    md="10"
    sm="10"
    style={
      Object {
        "paddingLeft": "0",
      }
    }
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
    xs="12"
  >
    <p
      className="addressTitle"
    >
      Alamat Tersimpan
    </p>
    <hr
      className="lineAddress"
    />
    <div
      key="34e6045b-5452-409d-cf3b-08d8bb5f8bb8"
    >
      <div
        className="addressBoxBilling"
      >
        <p
          className="address1"
        >
          Dwiky Fany Irawan
        </p>
        <p
          className="address2"
        >
          Dwiky Fany Irawan
        </p>
        <p
          className="address3"
        >
          Jl. Raya Bekasi No.KM.22, RT.7/RW.1, Cakung Bar., Kec. Cakung, Kota Jakarta Timur, Daerah Khusus Ibukota Jakarta 13910
        </p>
        <Row
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <p
            className="address4"
          >
            Edit Address
          </p>
          <p
            className="address5"
            onClick={[Function]}
          >
            Select Address
          </p>
        </Row>
      </div>
    </div>
    <div
      key="bbc23959-d18a-480e-cf3c-08d8bb5f8bb8"
    >
      <div
        className="addressBoxBillingSelected"
      >
        <p
          className="address1"
        >
          Kantor PT Global Service Indonesia
        </p>
        <p
          className="address2"
        >
          Indah Kurnia
        </p>
        <p
          className="address3"
        >
          Jl Raya kaliabang Tengah, Ruko San Jaya, Blok A5 No 9,
Sebalah Indomaret
        </p>
        <Row
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <p
            className="address4"
          >
            Edit Address
          </p>
        </Row>
      </div>
    </div>
    <div
      key="3eccd8ca-da3f-469f-cf3d-08d8bb5f8bb8"
    >
      <div
        className="addressBoxBilling"
      >
        <p
          className="address1"
        >
          Agit Mahenda
        </p>
        <p
          className="address2"
        >
          Agit Mahenda
        </p>
        <p
          className="address3"
        >
          Jl. Raya Randugarut No.Km. 12, Tugurejo, Kec. Tugu, Kota Semarang, Jawa Tengah 50186
        </p>
        <Row
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <p
            className="address4"
          >
            Edit Address
          </p>
          <p
            className="address5"
            onClick={[Function]}
          >
            Select Address
          </p>
        </Row>
      </div>
    </div>
    <div
      key="fafba700-cca2-44b6-cf3e-08d8bb5f8bb8"
    >
      <div
        className="addressBoxBilling"
      >
        <p
          className="address1"
        >
          Cuy
        </p>
        <p
          className="address2"
        >
          Cuy
        </p>
        <p
          className="address3"
        >
          string
        </p>
        <Row
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <p
            className="address4"
          >
            Edit Address
          </p>
          <p
            className="address5"
            onClick={[Function]}
          >
            Select Address
          </p>
        </Row>
      </div>
    </div>
    <div
      key="45e2a550-91f6-4103-ae40-08d8be27a682"
    >
      <div
        className="addressBoxBilling"
      >
        <p
          className="address1"
        >
          hallo guys
        </p>
        <p
          className="address2"
        >
          hallo guys
        </p>
        <p
          className="address3"
        >
          string
        </p>
        <Row
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <p
            className="address4"
          >
            Edit Address
          </p>
          <p
            className="address5"
            onClick={[Function]}
          >
            Select Address
          </p>
        </Row>
      </div>
    </div>
    <div
      key="f3acde09-8129-4dd9-5ef3-08d8c0dc1cc9"
    >
      <div
        className="addressBoxBilling"
      >
        <p
          className="address1"
        >
          Christiano Messi
        </p>
        <p
          className="address2"
        >
          Christiano Messi
        </p>
        <p
          className="address3"
        >
          Jalan Mars
        </p>
        <Row
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <p
            className="address4"
          >
            Edit Address
          </p>
          <p
            className="address5"
            onClick={[Function]}
          >
            Select Address
          </p>
        </Row>
      </div>
    </div>
    <p
      className="confirmAddress"
    >
      Confirm Address
    </p>
    <hr
      className="lineAddress"
    />
    <p
      className="addAddress"
      onClick={[Function]}
    >
      + Add New Address
    </p>
    <ModalAddress
      isClose={[Function]}
      isOpen={false}
    />
  </Col>
</div>
`;
